### PR TITLE
Remove noaa aws data install

### PR DESCRIPTION
**Pull request by biosafetylvl5**
* [x] Required ***existing tests*** pass (ie full_test.sh, others as appropriate)
* [x] NO REQUIRED ***unit tests*** (explain why not required)
* [x] NO REQUIRED ***integration tests*** (explain why not required)
* [x] NO REQUIRED ***documentation*** (explain why not required)
* [x] Required ***release notes*** added for new/modified functionality
* [x] NO REQUIRED ***updates to other repos*** (explain why not required)

# Related Issues
fixes NRLMMD-GEOIPS/geoips#764

# Testing Instructions
Freshly install geoips, run base install + test.

# Summary
Removed NOAA AWS dataset from the base tests because it was unused resulting in a ~6X faster install (2m 28s -> 24s) and a ~4.25 smaller data footprint (3.8Gb -> 894Mb).

### CODE REVIEW

**Comment by biosafetylvl5** [comment]
Install with noaa aws data: 2m 28s, 3.8GB
Install without:  24s, 894M

**Review by jsolbrig — APPROVED** [review]

**Review by mindyls — APPROVED** [review]

File: docs/source/releases/latest/764-integration-tests-pass-without-noaa-data.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+bug fix:
```
Comment (mindyls) [inline docs/source/releases/latest/764-integration-tests-pass-without-noaa-data.yaml:1]: Wait to merge until 1.14.0 is tagged/released.

Comment (biosafetylvl5) [reply docs/source/releases/latest/764-integration-tests-pass-without-noaa-data.yaml:1]: switched to dev-staging, so can merge now

**Review by jsolbrig — APPROVED** [review]

**Review by mindyls — APPROVED** [review]
No changes,  base branch changed to dev staging.  Reapproving

**Pull request merged**